src/targets/__tests__/maven.test.ts

```diff
@@ -0,0 +1,212 @@
+    );
+
+    const version = '1.0.0';
+    const revision = 'r3v1s10n';
+
+    await mvnTarget.publish(version, revision);
+    expect(gradlePropsMock).toHaveBeenCalledTimes(1);
+    expect(uploadMock).toHaveBeenCalledTimes(1);
+    expect(uploadMock).toHaveBeenLastCalledWith(revision);
+    expect(retrySpawnProcess).toHaveBeenCalledTimes(1);
+    expect(retrySpawnProcess).toHaveBeenCalledWith(DEFAULT_OPTION_VALUE, [
+      'closeAndReleaseRepository',
+    ]);
+    expect(callOrder).toStrictEqual([
+      'gradleProps',
+      'upload',
+      'closeAndRelease',
+    ]);
+  });
+
+  test('upload', async () => {
+    // simple mock to always use the same temporary directory,
+    // instead of creating a new one
+    (withTempDir as jest.MockedFunction<typeof withTempDir>).mockImplementation(
+      async cb => {
+        return await cb(tmpDirName);
+      }
+    );
+
+    const mvnTarget = createMavenTarget();
+    mvnTarget.getArtifactsForRevision = jest
+      .fn()
+      .mockResolvedValueOnce([{ filename: 'mockArtifact.zip' }]);
+    mvnTarget.artifactProvider.downloadArtifact = jest
+      .fn()
+      .mockResolvedValueOnce('artifact/download/path');
+
+    await mvnTarget.upload('r3v1s10n');
+    expect(retrySpawnProcess).toBeCalledTimes(1);
+    const callArgs = (retrySpawnProcess as jest.MockedFunction<
+      typeof retrySpawnProcess
+    >).mock.calls[0];
+
+    expect(callArgs).toHaveLength(2);
+    expect(callArgs[0]).toBeDefined();
+    expect(callArgs[0]).toEqual(DEFAULT_OPTION_VALUE);
+
+    const cmdArgs = callArgs[1];
+    expect(cmdArgs).toBeDefined();
+    expect(cmdArgs).toHaveLength(10);
+    // @ts-ignore `cmdArgs[*]` possibly undefined
+    expect(cmdArgs[0]).toEqual('gpg:sign-and-deploy-file');
+    // @ts-ignore `cmdArgs[*]` possibly undefined
+    expect(cmdArgs[1]).toMatch(new RegExp(`-Dfile=${tmpDirName}.*`));
+    // @ts-ignore `cmdArgs[*]` possibly undefined
+    expect(cmdArgs[2]).toMatch(
+      new RegExp(
+        `-Dfiles=${tmpDirName}.*-javadoc\.jar,${tmpDirName}.*-sources\.jar`
+      )
+    );
+    // @ts-ignore `cmdArgs[*]` possibly undefined
+    expect(cmdArgs[3]).toMatch(new RegExp(`-Dclassifiers=javadoc,sources`));
+    // @ts-ignore `cmdArgs[*]` possibly undefined
+    expect(cmdArgs[4]).toMatch(new RegExp(`-Dtypes=jar,jar`));
+    // @ts-ignore `cmdArgs[*]` possibly undefined
+    expect(cmdArgs[5]).toMatch(
+      new RegExp(`-DpomFile=${tmpDirName}.*pom-default\.xml`)
+    );
+    // @ts-ignore `cmdArgs[*]` possibly undefined
+    expect(cmdArgs[6]).toMatch(
+      new RegExp(`-DrepositoryId=${DEFAULT_OPTION_VALUE}`)
+    );
+    // @ts-ignore `cmdArgs[*]` possibly undefined
+    expect(cmdArgs[7]).toMatch(new RegExp(`-Durl=${DEFAULT_OPTION_VALUE}`));
+    // @ts-ignore `cmdArgs[*]` possibly undefined
+    expect(cmdArgs[8]).toMatch(new RegExp(`--settings`));
+    // @ts-ignore `cmdArgs[*]` possibly undefined
+    expect(cmdArgs[9]).toMatch(DEFAULT_OPTION_VALUE);
```

Comment:
I think these assertions can be a simple text equality with `toBe` or something?